Source/Engine/PackageBuilder/SyntaxParser.cs

```diff
@@ -1345,10 +1418,22 @@ private void ValidateToken(TokenId id, string error)
                 throw SyntaxError(fToken.TextSlice.Position, error, args: fToken);
         }
 
+        private void AttachSourceTextInformation(Syntax syntax, Token startToken)
+        {
+            syntax.StartLine = startToken.Line;
+            syntax.StartCharacter = startToken.Character;
+            syntax.EndLine = fLastToken.Line;
+            syntax.EndCharacter = fLastToken.Character + fLastToken.TextSlice.Length - 1;
+            int textSliceLength = fLastToken.TextSlice.End - startToken.TextSlice.Position + 1;
+            syntax.TextSlice = new Slice(fText, startToken.TextSlice.Position, textSliceLength);
+        }
+
         private struct Token
         {
             public TokenId Id;
             public Slice TextSlice;
+            public int Line { get; set; }
+            public int Character { get; set; }
```

Comment:
There is no need for line number and character position information in Token object.